Svc/Version/Version.fpp

```diff
@@ -0,0 +1,138 @@
+module Svc {
+    
+    @ Tracks versions for project, framework and user defined versions etc
+    
+    enum VersionEnabled {
+        DISABLED = 0 @< verbosity disabled
+        ENABLED = 1 @< verbosity enabled
+    }
+    
+   
+    @ An enumeration for Version Type
+    enum VersionType {
+        PROJECT = 0 @< project version
+        FRAMEWORK = 1 @< framework version
+        LIBRARY = 2 @< library version
+        CUSTOM = 3 @< custom version
+        ALL = 4 @< all above versions
+    }
+
+    @Data Structure for custom version Tlm
+    struct CustomVersionDb {
+        version_enum: VersionCfg.VersionEnum @<enumeration/name of the custom version
+        version_value: string size 80  @< string containing custom version
+        version_status : Svc.VersionStatus @< status of the custom version
```

Comment:
Yeah I debated about having this, in the end decided to leave it so it could be project specific. I figured this would carry more meaning for when "getVersion" is called rather than providing a downlink for the status